pkg/nextgen/distdoc/graph_test.go

```diff
@@ -0,0 +1,175 @@
+package distdoc
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/binary"
+	"github.com/nuts-foundation/nuts-network/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+// trackingVisitor just keeps track of which nodes were visited in what order.
+type trackingVisitor struct {
+	documents []Document
+}
+
+func (n *trackingVisitor) Accept(document Document, next []Document) {
+	n.documents = append(n.documents, document)
+}
+
+func (n trackingVisitor) JoinContentsAsString() string {
+	var contents []string
+	for _, document := range n.documents {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(document.Payload())
+		contents = append(contents, buf.String())
+	}
+	return strings.Join(contents, ", ")
+}
+
+func TestChain_Add(t *testing.T) {
+	key, certificate := generateKeyAndCertificate()
+	t.Run("ok", func(t *testing.T) {
+		graph := NewDAG()
+		doc := testDocument(0)
+		err := graph.Add(doc)
+		assert.NoError(t, err)
+		visitor := trackingVisitor{}
+		graph.Walk(visitor.Accept)
+		assert.Len(t, visitor.documents, 1)
+		assert.Equal(t, doc.Ref(), visitor.documents[0].Ref())
+	})
+	t.Run("ok - out of order", func(t *testing.T) {
+		_, documents := graphF(key, certificate)
+		graph := NewDAG()
+		for i := len(documents) - 1; i >= 0; i-- {
+			graph.Add(documents[i])
+		}
+		visitor := trackingVisitor{}
+		graph.Walk(visitor.Accept)
+		assert.Equal(t, "A, B, C, D, E, F", visitor.JoinContentsAsString())
+	})
+	t.Run("error - cyclic graph", func(t *testing.T) {
+		// A -> B -> C -> B
+		A := testDocument(0)
+		B := testDocument(1, A.Ref()).(*document)
+		C := testDocument(2, B.Ref())
+		B.prevs = append(B.prevs, C.Ref())
+
+		graph := NewDAG()
+		_ = graph.AddMultiple(A, B, C)
+		// TODO: Implement this check
```

Comment:
tests that are not ready yet can be skipped with t.Skip. This will show up and reminds us of todo's